### PR TITLE
Lower default maxRowsInMemory for realtime ingestion.

### PR DESCRIPTION
**Pull request by gianm**
The thinking with the current defaults is that for best ingestion throughput, we want intermediate persists to be as big as possible without using up all available memory. So, we rely mainly on maxBytesInMemory. The default maxRowsInMemory (1 million) is really just a safety: in case we have a large number of very small rows, we don't want to get overwhelmed by per-row overheads.

However, maximum ingestion throughput isn't necessarily the primary goal for realtime ingestion. Query performance is also important. And because query performance is not as good on the in-memory dataset, it's helpful to keep it from growing too large. 150k seems like a reasonable balance here. It means that for a typical 5 million row segment, we won't trigger more than 33 persists due to this limit, which is a reasonable number of persists.

### CODE REVIEW

File: server/src/main/java/org/apache/druid/segment/indexing/RealtimeTuningConfig.java [hunk 1]
```diff
@@ -129,7 +129,7 @@ public RealtimeTuningConfig(
   )
   {
     this.appendableIndexSpec = appendableIndexSpec == null ? DEFAULT_APPENDABLE_INDEX : appendableIndexSpec;
-    this.maxRowsInMemory = maxRowsInMemory == null ? DEFAULT_MAX_ROWS_IN_MEMORY : maxRowsInMemory;
+    this.maxRowsInMemory = maxRowsInMemory == null ? DEFAULT_MAX_ROWS_IN_MEMORY_BATCH : maxRowsInMemory;
```
Comment (kfaraz) [inline server/src/main/java/org/apache/druid/segment/indexing/RealtimeTuningConfig.java:132]: Probably got missed.
```suggestion
    this.maxRowsInMemory = maxRowsInMemory == null ? DEFAULT_MAX_ROWS_IN_MEMORY_REALTIME : maxRowsInMemory;
```

**Review by kfaraz — APPROVED** [review]
+1 after the CI passes.

There are some test failures:
```
Error:  Failures: 
Error:  org.apache.druid.indexing.common.task.TaskSerdeTest.testIndexTaskTuningConfigDefaults
Error:    Run 1: TaskSerdeTest.testIndexTaskTuningConfigDefaults:112 expected:<150000> but was:<1000000>
Error:    Run 2: TaskSerdeTest.testIndexTaskTuningConfigDefaults:112 expected:<150000> but was:<1000000>
Error:    Run 3: TaskSerdeTest.testIndexTaskTuningConfigDefaults:112 expected:<150000> but was:<1000000>
Error:    Run 4: TaskSerdeTest.testIndexTaskTuningConfigDefaults:112 expected:<150000> but was:<1000000>
```

**Comment by xvrl** [comment]
@gianm in the draft release notes we indicate this change improves throughput, but from reading your reasoning it sounds like this actually improves query performance at the expense of throughput, am I reading this right?

**Comment by clintropolis** [comment]
> @gianm in the draft release notes we indicate this change improves throughput, but from reading your reasoning it sounds like this actually improves query performance at the expense of throughput, am I reading this right? cc @clintropolis

yeah, this is optimizing for query perf by flushing to disk more frequently, fixed the release notes to say instead

>Optimized query performance by lowering the default maxRowsInMemory for real-time ingestion, which might lower overall ingestion throughput https://github.com/apache/druid/pull/13939

**Comment by gianm** [comment]
Yes, correct, the intent of this change is to adjust the balance of the defaults towards caring more about query perf.

**Pull request merged**